src/index.ts

```diff
@@ -446,7 +444,7 @@ function getValidCacheMap<K, V, C>(
 }
 
 // Private
-function isArrayLike(x: mixed): boolean {
+function isArrayLike(x: any): boolean {
```

Comment:
Is there any reason to not make this function a guard?

```suggestion
function isArrayLike(x: unknown): x is unknown[]  {
```

Reply:
no reason, I just ported Flow to TypeScript to make the PR as clean as possible, pretty much 1-to-1.
Your example will result in `values: ArrayLike<Error | V> & unknown[]` so I would use a generic here, rather than `unknown`.

Reply:
```suggestion
function isArrayLike<T>(x: any): x is ArrayLike<T> {
```

Reply:
I see. I wasn't thinking about what it would look like at the end. I was curious about the use of any given how unfriendly it is when using the library. `ArrayLike<T>` is, as you said, what makes the most sense here.

Reply:
It's internal function anyway